src/Common/Common.Authentication/WindowsAuthenticationFactory.cs

```diff
@@ -74,14 +75,68 @@ public IAccessToken Authenticate(IPowerBIEnvironment environment, IPowerBILogger
             return token.ToIAccessToken();
         }
 
-        private static TokenCache InitializeCache(IPowerBIEnvironment environment, string queryParams)
+        public IAccessToken Authenticate(IPowerBIEnvironment environment, IPowerBILogger logger, IPowerBISettings settings, string userName, SecureString password)
```

Comment:
This Authenticate method looks similar to the one above. Consider moving common behavior to private methods.

Reply:
Yes this could be refactored better